### ISSUE TITLE
Visual Studio files clutter the src directory and make maintenance harder.

### ISSUE THREAD

**Issue by sidneycadot** [cc65/cc65#2578]
Hi,

The src directory is now littered with a bunch of files that are there to allow Windows-based builds in the Github CI:

![image](https://github.com/user-attachments/assets/1eec7cc9-398e-460e-987d-92874ca28450)

All those vcxproj, sln, props, and cmd files clutter what used to be a pretty clean src/ directory.

Apart from aesthetics, this also presents a maintenance issue. There is no longer a single source of truth for dependencies (which used to be the Makefiles, which do most of the dependency analysis on the fly); but now, it turns out to be necessary to also hack in the Visual Studio files to get stuff to compile. Or perhaps it is even necessary to use Visual Studio and its project management functionality to update these files?

Here's a Windows CI build log that I'm getting, even after adding those new source files to the "vcxproj" files, that I have no idea how to solve.

This is while working on PR #2577, which adds an extra file peripherals.c to src/sim65. Fortune favors the brave, so I added that file (and its header file) to the src/sim65.vcxproj file; but still, the build fails.

![image](https://github.com/user-attachments/assets/9c0596f3-5d89-4bee-8708-9c43c656dd16)

For some reason (a bit earlier in the file), the compiler doesn't compile peripherals.c

![image](https://github.com/user-attachments/assets/728f2578-4d02-4179-835d-4038962f71fa)

This is weird. It is completely unclear to me why the Windows compiler does not compile peripherals.c, even after adding it to  src/sim65.vcxproj. How does the Windows build decide which files to compile?

I can see some added value in a windows-based build with a different compiler and linker, but the price we pay for that (in terms of having to clutter the src directory, and having to maintain opaque vcxproj files) is just too high for my taste.

**Comment by mrdudz** [comment]
One of the main reasons those do even still exist was that @acqn - at that time the most active developer - was using windows only.

The reason for me to have them included in CI was: if they are not, then they will stop working soon. So if we don't decide to remove them alltogether, they'll have to be part of CI, one way or another.

**Comment by mrdudz** [comment]
> I don't understand what you're saying here, sorry. :)

We could keep the files around, but not use them in CI - meaning they will be broken most of the time, which pretty much defeats the purpose of even having them at all. So to me it boils down to either fix/maintain those files, or to remove them (and provide a mingw based windows build). (*)

That said, keeping them obviously needs someone to do what you said (not me - @acqn used to do this). @oliverschmidt was a working on windows iirc, i don't recall who else was though. @spiro-trikaliotis moved to linux i think? :)

I have to admit that i never saw this problem - ie a developer working on linux is adding files. Obviously this can not be a problem in any sane workflow, so we need to fix this.

 (*) i would accept a patch that moves the existing msvc stuff into a separate directory, plus adds a readme into that directory that explains a) that they are not maintained and b) how to update them using msvc

**Comment by sidneycadot** [comment]
Whatever we're going to do, I'd like to do it sooner rather than later. Progress on #2577 is currently blocked because of it: the Windows-based CI build there fails.

I would happily make a PR to wipe these files from existence (I'm not much of a Microsoft fan, and this to me is a typical example how nearly everything they touch is somewhat badly done), but that's all I can do. If someone would want to step up and be the go-to person to keep the Windows build stuff working, that'd be great, but if that's not going to happen, we'd be better off getting rid of this stuff ASAP.

I guess it's your call as to how long we wait to pull the trigger.

**Comment by oliverschmidt** [comment]
1. I personally don't need/use the files in question anymore.

2. The files in question are - of course - not relevant for just building `src`. They become relevant when using the - superior - VS debugger. So the ones to decide here should be the ones actually working on `src`. If they want/need those files I personally consider this (way) more relevant than anybody else being unhappy about them - for whatever reason.

3. Since many years the only relevant changes done to `src` is to add (or remove) a file here and there. Those changes are easily applied to the files in question - way easier than (re)creating those files from scratch. Therefore I would vote to not remove them.

Just my two cents, Oliver

**Comment by sidneycadot** [comment]
Hi @oliverschmidt,

I am myself quite adamantly in opposition of putting IDE-specific project files in a source directory. They are essentially opaque files, not source files. I also think it's rather a hefty price to pay to pollute the src/ directory for some people wanting to use VS.

A better way for the use case you mention would be to collect these project files in some "misc/ide/visual_studio" directory, where they would be the responsibility of whoever likes to use them, with all the benefits and obligations (like keeping them in sync) that go with that.

The problem before us is that these files are not only used to facilitate people wanting to use VS, but they also appear to be an essential part of the Windows-based CI build. And when that breaks and people (like myself) don't know how to fix it, that's a problem.

> Those changes are easily applied to the files in question

A cursory look at https://github.com/cc65/cc65/blob/master/.github/workflows/build-on-pull-request.yml suggests that the Windows-based build depends on src/cc65.sln, which refers to src/sim65.vcxproj.

I tried to add src/sim65/peripheral.[ch] to src/sim65.vcxproj (see: https://github.com/cc65/cc65/pull/2577/files), and still the workflow build fails. 

So what should I do? Can this be done without Windows and Visual Studio? (If not, I wouldn't call it "easy".)

**Comment by oliverschmidt** [comment]
Thanks for the clarification! I personally don't think it's desirable to have a CI based on those files in the first place - for the very reasons you just gave.

**Comment by sidneycadot** [comment]
Right, the compilation problem I encountered while manually editing the sim65.vcxproj was resolved, thanks to a code review by sharp eyed Bloch sphere juggler @matpompili. Turns out I made a silly mistake, adding the peripherals.c file using a 'ClInclude' instead of a 'ClCompile' directive.

The point stands, more or less -- editing these files by hand (and not quite knowing what you're doing) is a pretty fragile way of working.

**Comment by spiro-trikaliotis** [comment]
> @spiro-trikaliotis moved to linux i think?

My last Visual Studio was Version 6, so I am not the right person for this here.
Although from time to time, I install a newer version in a VM for testing purposes if I need it, but if I can avoid it, I don't do it.

**Comment by mrdudz** [comment]
I'll have a final comment and close :)

First of all i have to say that from my perspective, all of this windows specific stuff should die in a dumpster fire. That includes all the weird magic @oliverschmidt added to the Makefiles to support cmd.exe

However, i totally understand that these things are useful for windows users (and the MSVC debugger IS superior indeed). Which is why these things still exist, and should stay there. (@acqn used it, for example - and we would have missed a bunch of useful contributions without it)

Now, what i *really* hate is dragging along bitrot - which is why in my book it is not only a good thing, but even a must, to have all those things included in CI - because if things are not tested, they will break (and we had this before we added the CI stuff for them). Ultimatively it saves time and hazzle for everyone - even if editing those MSVC files or understanding the Makefile magic is not straightforward (And i actually learned some interesting things from the later - and i have adopted some things for my own Projects).

So, from my point of view, its either "all of it gets cleaned out completely" or "it stays there and is used in CI". I don't mind what it is - but i DO mind untested garbage files.

**Comment by sidneycadot** [comment]
Strange that you chose to close this. I don't consider this issue resolved.
